### PR TITLE
Stellar integration

### PR DESCRIPTION
**Pull request by yosriz**
First PR for stellar integration:

* replace geth with stellar sdk
* working transaction sending + balance 
* remove not relevant classes
* TBD - testing (unit + integ) , keystore (currently saved unsecure in sharedpref), enhanced error handling, remove pending balance, onboarding (change kin trust, friend bot etc.), docs, ci, distribution (aar), entire scope can be seen here: https://medium.com/p/839afcaaf55c

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -26,6 +26,7 @@ allprojects {
         maven {
             url "https://dl.bintray.com/kinfoundation/go-ethereum"
```
Comment (OrenZak) [inline build.gradle]: we can remove that?

File: kin-sdk-core/src/androidTest/assets/testConfig.json [hunk 2]
```diff
@@ -30,5 +30,5 @@
  "private_key":"0xabf8c2dd52f5b14ea437325854048e5daadbca80f99f9d6f8e97ab5e05d4f0ab"
  }
 ], 
-"token_contract_address":"0x8919486b0afaad4656e8f9667ce9adbb62c3f2b1"
+"token_contract_address":""
```
Comment (OrenZak) [inline kin-sdk-core/src/androidTest/assets/testConfig.json]: we can add this file into .gitignore

File: kin-sdk-core/src/main/java/kin/sdk/core/EncryptedAccount.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package kin.sdk.core;
+
+
+class EncryptedAccount {
```
Comment (OrenZak) [inline kin-sdk-core/src/main/java/kin/sdk/core/EncryptedAccount.java]: maybe change to just: Account?

Comment (yosriz) [reply kin-sdk-core/src/main/java/kin/sdk/core/EncryptedAccount.java]: Naming is hard... it's pretty lame one, Maybe `StellarAccount`?

File: build.gradle (hunk 1 above)
Comment (yosriz) [reply build.gradle]: happily

File: kin-sdk-core/src/main/java/kin/sdk/core/EncryptedAccount.java (hunk 3 above)
Comment (OrenZak) [reply kin-sdk-core/src/main/java/kin/sdk/core/EncryptedAccount.java]: Simple as possible that's what I think, `Account` same as we have on ethereum sdk

File: kin-sdk-core/src/androidTest/assets/testConfig.json (hunk 2 above)
Comment (yosriz) [reply kin-sdk-core/src/androidTest/assets/testConfig.json]: I will get rid of this files soon when I get to tests

File: kin-sdk-core/src/main/java/kin/sdk/core/EncryptedAccount.java (hunk 3 above)
Comment (yosriz) [reply kin-sdk-core/src/main/java/kin/sdk/core/EncryptedAccount.java]: the problem is we have KinAccount, I want to make a distinction between this objects, geth has their own `Account` on their API

Comment (OrenZak) [reply kin-sdk-core/src/main/java/kin/sdk/core/EncryptedAccount.java]: I still think it should be `Account` :)
👍

Comment (yosriz) [reply kin-sdk-core/src/main/java/kin/sdk/core/EncryptedAccount.java]: ok, I'll make it simple

File: kin-sdk-core/src/androidTest/assets/testConfig.json (hunk 2 above)
Comment (OrenZak) [reply kin-sdk-core/src/androidTest/assets/testConfig.json]: 👍  thanks

File: kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java [hunk 4]
```diff
@@ -25,7 +22,7 @@
      * smart-contract problems.
      */
     public KinClient(Context context, ServiceProvider provider) throws EthereumClientException {
```
Comment (OrenZak) [inline kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java]: We need to change `EthereumClientException` to be more generic like `ClientException`

File: kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java [hunk 5]
```diff
@@ -105,7 +91,7 @@ public void deleteAccount(String passphrase) throws DeleteAccountException {
      * WARNING - if you don't export your account before deleting it, you will lose all your Kin.
      */
     public void wipeoutAccount() throws EthereumClientException {
```
Comment (OrenZak) [inline kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java]: same here `EthereumClientException`

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        initKeyStore();
+    }
+
+    @NonNull
+    KeyStore getKeyStore() {
+        return keyStore;
+    }
+
+    @NonNull
+    TransactionId sendTransaction(@NonNull Account from, @NonNull String publicAddress,
```
Comment (OrenZak) [inline kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: We don't want to do any authentication with `passphrase`?
Something like: get the secret seed only with the correct `passphrase`?

File: kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java (hunk 4 above)
Comment (yosriz) [reply kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java]: indeed, already done at a different [branch](https://github.com/kinfoundation/kin-sdk-core-stellar-android/blob/error_handling_enhancements/kin-sdk-core/src/main/java/kin/sdk/core/exception/ClientException.java)

File: kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java (hunk 5 above)
Comment (yosriz) [reply kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java]: same as above

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java (hunk 6 above)
Comment (yosriz) [reply kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: thanks, lost in translation.. fixed now

File: kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java (hunk 4 above)
Comment (OrenZak) [reply kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java]: 👍

File: kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java (hunk 5 above)
Comment (OrenZak) [reply kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java]: 👍

File: build.gradle [hunk 7]
```diff
@@ -23,9 +23,7 @@ allprojects {
         flatDir {
             dirs project(projectName).file('libs')
```
Comment (OrenZak) [inline build.gradle]: i think we can remove also this block of `flatDir`

**Comment by YohayBar** [comment]
@yosriz can we clean erc20 history - i.e copy starting point into new folder, create new repo and push starting point,  delete this repo.

I don't think that who ever look or clone this repo should have code history pre the stellar usage

**Comment by yosriz** [comment]
@YohayBar, the idea was to 'fork' the existing one, and change the code based on previous work, is that really matters we have the old history? anyhow I will not start from clean slate as the work is incremental.

**Comment by yosriz** [comment]
@OrenZak @YohayBar , should I rename the lib to `kin-sdk-core-stellar` to prevent future conflicts?

File: build.gradle (hunk 7 above)
Comment (yosriz) [reply build.gradle]: fixed

File: kin-sdk-core/src/main/java/kin/sdk/core/Account.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
+package kin.sdk.core;
+
+
+class Account {
```
Comment (YohayBar) [inline kin-sdk-core/src/main/java/kin/sdk/core/Account.java]: please add javadoc documentation - what is Account encryptedSeed, accountID etc.

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+        server = initServer();
+        keyStore = initKeyStore();
+        KeyPair issuer = KinConsts.getKinIssuer(serviceProvider);
+        Asset kinAsset = Asset.createNonNativeAsset(KinConsts.KIN_ASSET_CODE, issuer);
+        transactionSender = new TransactionSender(server, keyStore, issuer, kinAsset);
+    }
+
+    private Server initServer() {
+        if (serviceProvider.isMainNet()) {
+            Network.usePublicNetwork();
```
Comment (YohayBar) [inline kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: I think this logic should be within the service provider

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+        } else {
+            Network.useTestNetwork();
+        }
+        return new Server(serviceProvider.getProviderUrl());
```
Comment (YohayBar) [inline kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: also the server provider need to to provide stellar service (IMO)

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    /**
+     * Create {@link KeyStore}, to have control over the account management.
+     * And the ability to store accounts securely according to go-ethereum encryption protocol.
+     * The keystore path is unique to each network id,
+     * for example Ropsten network will be: ../data/kin/keystore/3/
```
Comment (YohayBar) [inline kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: is this true anymore?

File: kin-sdk-core/src/main/java/kin/sdk/core/KinConsts.java [hunk 12]
```diff
@@ -1,61 +1,38 @@
...
-
-    static long getTransferKinGasLimit(ServiceProvider provider) {
-        return NetworkConstants.fromProvider(provider).transferKinGasLimit;
-    }
+    static final String KIN_ASSET_CODE = "KIN";
 
-    static String getContractAddress(ServiceProvider provider) {
-        return NetworkConstants.fromProvider(provider).contractAddress;
+    static KeyPair getKinIssuer(ServiceProvider provider) {
+        return NetworkConstants.fromProvider(provider).issuer;
```
Comment (YohayBar) [inline kin-sdk-core/src/main/java/kin/sdk/core/KinConsts.java]: also can be part of ServiceProvider

File: kin-sdk-core/src/main/java/kin/sdk/core/KinConsts.java [hunk 13]
```diff
@@ -1,61 +1,38 @@
...
-    /**
-     * <p>SHA3 (Keccak-256) hash of ERC-20's Transfer event:</p>
-     * {@code Transfer(address,address,uint256)}
-     */
-    static final String TOPIC_EVENT_NAME_SHA3_TRANSFER = "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef";
-
-    static long getTransferKinGasLimit(ServiceProvider provider) {
-        return NetworkConstants.fromProvider(provider).transferKinGasLimit;
-    }
+    static final String KIN_ASSET_CODE = "KIN";
```
Comment (YohayBar) [inline kin-sdk-core/src/main/java/kin/sdk/core/KinConsts.java]: should file contacts file only include static final strings/init etc and not end, or static methods?

File: kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+    private final Server server;
+    private final KeyStore keyStore;
+    private final KeyPair issuer;
+    private final Asset kinAsset;
+
+    TransactionSender(Server server, KeyStore keyStore, KeyPair issuer, Asset kinAsset) {
+        this.server = server;
+        this.keyStore = keyStore;
+        this.issuer = issuer;
+        this.kinAsset = kinAsset;
```
Comment (YohayBar) [inline kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java]: should assets include the issuer as part of the asset information? https://www.stellar.org/developers/guides/concepts/assets.html

File: kin-sdk-core/src/main/java/kin/sdk/core/Account.java [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+
+    private final String encryptedData;
+    private final String accountId;
+
+    Account(String encryptedSeed, String accountId) {
+        this.encryptedData = encryptedSeed;
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/Account.java]: change to encryptedSeed

File: kin-sdk-core/src/main/java/kin/sdk/core/ServiceProvider.java [hunk 16]
```diff
@@ -3,38 +3,23 @@
...
      * <p>
-     * For example to connect to an infura test node use
-     * new ServiceProvider("https://ropsten.infura.io/YOURTOKEN", NETWORK_ID_ROPSTEN);
-     *
      * @param providerUrl the provider to use
-     * @param networkId for example see {@value #NETWORK_ID_MAIN} {@value NETWORK_ID_ROPSTEN} {@value
-     * NETWORK_ID_RINKEBY}
+     * @param networkId for example see {@value #NETWORK_ID_MAIN} {@value NETWORK_ID_TEST}
      */
     public ServiceProvider(String providerUrl, int networkId) {
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/ServiceProvider.java]: enum or @IntDef

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+     *
+     * @throws EthereumClientException if could not create directory to save the keystore.
+     */
+    private KeyStore initKeyStore() throws EthereumClientException {
+        return new KeyStore(context);
+    }
+
+    private String getKeyStorePath() {
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: remove

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+
+    ClientWrapper(Context context, ServiceProvider serviceProvider)
+        throws EthereumClientException {
+        this.serviceProvider = serviceProvider;
+        this.context = context.getApplicationContext();
+        server = initServer();
+        keyStore = initKeyStore();
+        KeyPair issuer = KinConsts.getKinIssuer(serviceProvider);
+        Asset kinAsset = Asset.createNonNativeAsset(KinConsts.KIN_ASSET_CODE, issuer);
+        transactionSender = new TransactionSender(server, keyStore, issuer, kinAsset);
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: use KinAsset for getting issuer

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+            .append("kin")
+            .append(File.separator)
+            .append("keystore")
+            .append(File.separator)
+            .append(serviceProvider.getNetworkId())
+            .toString();
+    }
+
+    void wipeoutAccount() throws EthereumClientException {
+        File keystoreDir = new File(getKeyStorePath());
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: delete and add TODO

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java [hunk 20]
```diff
@@ -0,0 +1,134 @@
...
+     * @param account the {@link KeyPair} to check balance
+     * @return the account {@link Balance}
+     * @throws OperationFailedException if could not retrieve balance
+     */
+    Balance getBalance(Account account) throws OperationFailedException {
+        Balance balance = null;
+        try {
+            AccountResponse accountResponse = server.accounts().account(KeyPair.fromAccountId(account.getAccountId()));
+            for (AccountResponse.Balance assetBalance : accountResponse.getBalances()) {
+                if (KinConsts.KIN_ASSET_CODE.equals(assetBalance.getAssetCode())) {
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: also check issuer ID, encapsulate checking for Kin asset

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java [hunk 21]
```diff
@@ -0,0 +1,134 @@
...
+        return balance;
+    }
+
+
+    ServiceProvider getServiceProvider() {
+        return serviceProvider;
+    }
+
+    private void deleteRecursive(File fileOrDirectory) {
+        if (fileOrDirectory.isDirectory()) {
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: delete

File: kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java [hunk 22]
```diff
@@ -114,15 +100,18 @@ public void wipeoutAccount() throws EthereumClientException {
...
+        return clientWrapper.getServiceProvider();
     }
 
     KinAccount importAccount(String privateEcdsaKey, String passphrase) throws OperationFailedException {
-        Account account = ethClient.importAccount(privateEcdsaKey, passphrase);
         KinAccount kinAccount = null;
-        if (account != null) {
-            kinAccount = new KinAccountImpl(ethClient, account);
-        }
+     /*   try {
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java]: remove and add TODO

**Comment by yosriz** [comment]
remove default proguard file (check if library should not have this)

**Pull request closed without merging**